qdrant-landing/content/blog/qdrant-relari.md

```diff
@@ -0,0 +1,289 @@
+from langchain_qdrant import Qdrant
+from langchain_community.embeddings.fastembed import FastEmbedEmbeddings
+from relari.core.types import DatasetDatum
+```
+
+### Load and chunk data
+
+We will use LangChain to prepare our data.
+
+```python
+# load the document and split it into chunks
+loader = DirectoryLoader("gitlab_legal_policies/")
+documents = loader.load_and_split()
+```
+
+Now we’ll use FastEmbed, Qdrant's built-in embedding provider, to embed our chunks.
+
+```python
+# Initialize FastEmbedEmbeddings
+embeddings = FastEmbedEmbeddings(
+    model_name="BAAI/bge-small-en-v1.5",  # specify the model
+)
+```
+
+### Store data in Qdrant
+
+Finally, we'll upload the chunks into a Qdrant collection.
+
+```python
+# Load chunks into a Qdrant Cloud vectorstore using FastEmbedEmbeddings
+db = Qdrant.from_documents(
+    documents,
+    embedding=embeddings,
+    url=os.environ['QDRANT_URL'],  # Qdrant Cloud URL
+    api_key=os.environ['QDRANT_API_KEY'],  # Qdrant Cloud API Key
+    collection_name="gitlab_legal_policies",
+)
+print(f"{len(documents)} chunks loaded into Qdrant Cloud vector database.")
+```
+<aside role="status">
+If you're using the GitLab dataset, you should have 66 chunks.
+</aside>
+
+### Start logging results
+
+Once we've uploaded our data, we'll build a function to run different RAG pipelines over the dataset and log the results.
+
+```python
+# Prepare a function to run different RAG pipelines over the dataset and log the results
+def log_retriever_results(retriever, dataset):
+  log = list()
+  for datum in dataset.data:
+      # First compute the result
+      retrieved_docs = retriever.invoke(datum["question"])
+      # Now log the result in Relari format
+      result = DatasetDatum(
+          label=datum["uid"],
+          data={"retrieved_context": [doc.page_content for doc in retrieved_docs]},
+      )
+      log.append(result)
+  return log
+```
+
+This is the power of combining Qdrant and Relari. Instead of having to build multiple applications, slowly upsert, and retrieve data, you can use both to quickly test different parameters and instantly get results. This evaluation system is built for fast, useful iteration.
+
+### Evaluate results
+
+Now that we have our application built, let's explore how to evaluate Top-K.
+
+#### First attempt
+
+Run an experiment with Top-K, meaning results shown to users will vary between 3, 5, 7, and 9.
+
+```python
+k_values = [3, 5, 7, 9]  # Define the different values of top k to experiment
+
+semantic_retrievers = {}
+semantic_logs = {}
+
+# Run the retrievers on the dataset and log retrieved chunks
+for k in k_values:
+    retriever = db.as_retriever(search_type="similarity", search_kwargs={"k": k})
+    log = log_retriever_results(retriever, dataset)
+
+    semantic_retrievers[f"k_{k}"] = retriever
+    semantic_logs[f"k_{k}"] = log
+    print(f"Results on {dataset.name} by Semantic Retriever with k={k} saved!")
+```
```

Comment:
Could you add the output of the code snippets to the post? If someone wants to read the article without running the code, they have no clue about the results.